src/common/evaluable.ts

```diff
@@ -46,6 +49,10 @@ export interface Evaluable {
    */
   evaluate(ctx: Context): Result;
 
+  simplify(ctx: Context): Result | Evaluable
```

Comment:
Could we drop here a comment, what exactly is the "simplify" method desired outcome? To help others to quickly get the idea of the intention

Reply:
I will. It is part of my docs task.

Reply:
I missed the TODO :)